### PR TITLE
No MK, No heroes

### PR DESCRIPTION
**Pull request by kobisjeruk**
First time using github, hopefully I'm doing it right. This is my edited version (please refer README.md).
Changes :-
main.ahk 
-remove hero check, trim down the time and change the hotkeys for start (Z) and stop/reload (X)

utils/towers.ahk
-change unit from ninja -> druid

maps
-revamped all maps

### CODE REVIEW

**Comment by Valokoodari** [comment]
I have not yet tested this properly but just by skimming through the changes this seems really good for the most part.

However, I did notice some code formatting problems (mostly mixed tabs and spaces, and trailing whitespace) which will need to be fixed before merging. Also the changes made in a63a1b0 will not be merged aside from the removal of the hero check as the hotkeys should not be anything common like cut and undo and the sleep times have been chosen to not spam the log too much. Even the hero check removal should be done properly instead of just commenting out the function call.

I will likely have time tomorrow or the day after that so I'll fix all these problems then. Also please name your commits properly in the future if you make any more changes as "Add files via upload" doesn't describe the changes made at all.

File: README.md [hunk 1]
```diff
@@ -42,18 +38,15 @@ Testing of the script is (hopefully) always live at [Twitch](https://www.twitch.
 <summary><h4>Required tower upgrades</h4></summary>
 
 - Dart Monkey 024
-- Tack Shooter 402
-- Sniper Monkey 204
-- Monkey Sub 404
-- Monkey Buccaneer 322
-- Monkey Ace 223
+- Sniper Monkey 302
```
Comment (Valokoodari) [inline README.md:41]: This should be at least 322

File: maps/bloody_puddles.ahk [hunk 2]
```diff
@@ -1,55 +1,51 @@
...
-    Upgrade(1186,193,0,0,1)     ; Sub: 201 -> 202
+    WaitForRound(4)             ; Round 4
+    Upgrade(1180,190,0,0,1)     ; Sub: 000 -> 001
 
-    WaitForRound(1)             ; Round 11
-    Upgrade(1186,193,0,0,1)     ; Sub: 202 -> 203
-    Place("sniper",835,180)     ; Place Sniper
-    Targeting(835,180,3)        ; Sniper: First -> Strong
+    WaitForRound(6)             ; Round 6
+    Upgrade(1180,190,2,0,0)     ; Sub: 000 -> 200
```
Comment (Valokoodari) [inline maps/bloody_puddles.ahk:13]: There's something wrong here

**Review by Valokoodari — CHANGES_REQUESTED** [review]
Every map has now been tested successfully at least once so this is almost done. Just some small fixes required.

**Review by Valokoodari — APPROVED** [review]
Just going to test this for a few more hours but otherwise everything should be fine now.

**Comment by Valokoodari** [comment]
Result so far are 175 victories and 14 defeats which isn't quite as good as the older strategies (~1000 victories to 10 defeats). I still think the results are good enough so I'm merging this now.

**Comment by kobisjeruk** [comment]
Honestly, more than anything else I am quite shocked to see that 14 defeats. Were any particular map prove to be problematic? I wish I could reproduce the result but I am facing a different problem (not really a problem but sometimes script got randomly stuck at the victory screen).

**Comment by Valokoodari** [comment]
Currently at 414 victories and 28 defeats. All of the defeats have been on either Ravine (9) or Workshop (19). On Workshop the problem is clear: sometimes too many bloons leak and the script cannot afford the spike factory. On Ravine I haven't yet seen what the problem is but I suspect it to be something similar.

**Comment by kobisjeruk** [comment]
I suspected it is something related to timing. Almost always I tried to have some leeway before purchase/upgrade. Mostly if the purchase or upgrade use up almost all cash (like say <$50 left), I'd wait until the next round instead. Ravine is one where it is quite 'tight'. I didnt realize it happened in Workshop. A simple solution is to bump the purchase/upgrade to the next round. I guess you fix it already?

**Comment by Valokoodari** [comment]
Yeah, I fixed them already in commits 1db789cac5d1cfb5541cf5d36adaa378349a9388 and a3c755959adc728b818fc84b839eae12d264d116.

**Pull request merged**